src/autopas/containers/linkedCells/traversals/C08LikeBaseCellProcessor.h

```diff
@@ -0,0 +1,146 @@
+/**
+ * @file C08LikeBaseCellProcessor.h
+ * @author S. Seckler
+ * @date 10.01.2019
+ */
+
+#pragma once
+
+#include "autopas/containers/cellPairTraversals/CellPairTraversal.h"
+#include "autopas/pairwiseFunctors/CellFunctor.h"
+#include "autopas/utils/ThreeDimensionalMapping.h"
+
+namespace autopas {
+
+/**
+ * This class provides the base for traversals using the c08 base step.
+ *
+ * The base step processBaseCell() computes one set of pairwise interactions
+ * between two cells for each spatial direction based on the baseIndex.
+ * After executing the base step on all cells all pairwise interactions for
+ * all cells are done.
+ *
+ * @tparam ParticleCell the type of cells
+ * @tparam PairwiseFunctor The functor that defines the interaction of two particles.
+ * @tparam useSoA
+ * @tparam useNewton3
+ */
+template <class ParticleCell, class PairwiseFunctor, bool useSoA, bool useNewton3>
+class C08LikeBaseCellProcessor {
+ public:
+  /**
+   * Constructor of the c08 traversal.
+   * @param pairwiseFunctor The functor that defines the interaction of two particles.
+   */
+  explicit C08LikeBaseCellProcessor(PairwiseFunctor *pairwiseFunctor, std::array<unsigned long, 3> cellsPerDimension)
+      : _cellFunctor(
+            CellFunctor<typename ParticleCell::ParticleType, ParticleCell, PairwiseFunctor, useSoA, useNewton3>(
+                pairwiseFunctor)),
+        _cellPairOffsets{},
+        _cellOffsets{} {
+    computeOffsets(cellsPerDimension);
+  }
+
+  /**
+   * Computes one interaction for each spacial direction based on the lower left
+   * frontal corner (=base index) of a 2x2x2 block of cells.
+   * @param cells vector of all cells.
+   * @param baseIndex Index respective to which box is constructed.
+   */
+  void processBaseCell(std::vector<ParticleCell> &cells, unsigned long baseIndex);
+
+ private:
+  /**
+   * Computes pairs for the block used in processBaseCell()
+   * @param cellsPerDimension
+   */
+  void computeOffsets(std::array<unsigned long, 3> cellsPerDimension);
+
+  /**
+   * CellFunctor to be used for the traversal defining the interaction between two cells.
+   */
+  CellFunctor<typename ParticleCell::ParticleType, ParticleCell, PairwiseFunctor, useSoA, useNewton3> _cellFunctor;
```

Comment:
In some traversals, you call this `_cellFunctor` and in some `_cellProcessor` -> consistency plz :)

Reply:
invalid.
_cellFunctor is an actual cell functor
_baseCellProcessor handles the processBaseCell, which is built upon traversals.